src/realm/views.hpp

```diff
@@ -56,20 +56,53 @@ class SortDescriptor {
     {
         return !m_columns.empty();
     }
-
-    // handover support
-    using HandoverPatch = std::unique_ptr<SortDescriptorHandoverPatch>;
-    static void generate_patch(SortDescriptor const&, HandoverPatch&);
-    static SortDescriptor create_from_and_consume_patch(HandoverPatch&, Table const&);
-
+    // returns whether this descriptor has any custom ascending/descending order
+    bool has_custom_order() const noexcept
+    {
+        return std::any_of(m_ascending.begin(), m_ascending.end(), [](bool b) { return !b; });
```

Comment:
Good point, `has_any_descending_order()` would probably be more descriptive of what it actually returns. I can change it. Though this would be entirely unnecessary if distinct had compile time safety of not adding any ascending order. Though it might be a heavy weight solution, I'm going to consider changing the design to having separate classes for SortDescriptor and DistinctDescriptor again.